geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/ClientSNIAcceptanceTest.java

```diff
@@ -58,50 +64,143 @@
+
+    // set up client cache properties so it can connect to the server
+    Properties clientCacheProperties = new Properties();
+    clientCacheProperties.setProperty(SSL_ENABLED_COMPONENTS, "all");
+    clientCacheProperties.setProperty(SSL_KEYSTORE_TYPE, "jks");
+    clientCacheProperties.setProperty(SSL_REQUIRE_AUTHENTICATION, "false");
+
+    clientCacheProperties.setProperty(SSL_TRUSTSTORE, trustStorePath);
+    clientCacheProperties.setProperty(SSL_TRUSTSTORE_PASSWORD, "geode");
+    clientCacheProperties.setProperty(SSL_ENDPOINT_IDENTIFICATION_ENABLED, "true");
+    cache = getClientCache(clientCacheProperties);
+
+    // the gfsh startup script created a server-side region named "jellyfish"
+    region = cache.<String, String>createClientRegionFactory(ClientRegionShortcut.PROXY)
+        .create("jellyfish");
+    bulkData = getBulkDataMap();
+    region.putAll(bulkData);
+  }
+
+  @After
+  public void after() {
+    // preserve this commented code for debugging
+    // String logs = docker.exec(options("-T"), "geode",
+    // arguments("cat", "server-dolores/server-dolores.log"));
+    // System.out.println("server logs------------------------------------------");
+    // System.out.println(logs);
+
+    if (cache != null) {
+      cache.close();
+      cache = null;
+    }
+    bulkData = null;
+    region = null;
   }
 
+  // run all tests in one @Test to avoid having to stage a new Docker cluster for each
   @Test
+  public void runAllTests() throws Exception {
+    connectToSNIProxyDocker();
+    verifyServerAPIs();
+    query();
+    getAll();
+    removeAll();
+  }
+
+  /**
+   * A basic connectivity test that performs a few simple operations
+   */
   public void connectToSNIProxyDocker() {
-    Properties gemFireProps = new Properties();
-    gemFireProps.setProperty(SSL_ENABLED_COMPONENTS, "all");
-    gemFireProps.setProperty(SSL_KEYSTORE_TYPE, "jks");
-    gemFireProps.setProperty(SSL_REQUIRE_AUTHENTICATION, "false");
+    region.put("hello", "world");
+    assertThat(region.containsKey("hello")).isFalse(); // proxy regions don't store locally
+    assertThat(region.get("hello")).isEqualTo("world");
+    region.destroy("hello");
+    assertThat(region.get("hello")).isNull();
+  }
+
+  /**
+   * A test of Region query that returns a "big" result
+   */
+  public void query() throws Exception {
+    final SelectResults<String> results = region.query("SELECT * from /jellyfish");
+    assertThat(results).hasSize(bulkData.size());
+    for (String result : results) {
+      assertThat(bulkData.containsValue(result)).isTrue();
+    }
+  }
+
+  /**
+   * A test of Region bulk getAll
+   */
+  public void getAll() {
+    final Map<String, String> results = region.getAll(bulkData.keySet());
+    assertThat(results).hasSize(bulkData.size());
+    for (Map.Entry<String, String> entry : results.entrySet()) {
+      assertThat(region.containsKey(entry.getKey())).isFalse();
```

Comment:
The local region is a PROXY.  It's empty, so it won't have the key.   I put in that check in case someone accidentally changes the type of region in the client.  It would invalidate a lot of the tests if the client cache was stateful.

Reply:
nice